### PR TITLE
chore(deps): Upgrade dependencies to latest

### PR DESCRIPTION
**Pull request by r-near**
* Upgrades all dependencies to the latest
* Fixes deprecated use of `try_to_vec`
* Supports new `wait_until` parameter

### CODE REVIEW

File: near-fetch/src/query.rs
Comment (r-near) [inline near-fetch/src/query.rs:602]: Is this the right wait_until status to use?

File: near-fetch/src/query.rs
Comment (r-near) [inline near-fetch/src/query.rs:612]: Not sure what to do here

File: near-fetch/src/ops.rs
Comment (r-near) [inline near-fetch/src/ops.rs:315]: VS Code suggested this change

File: near-fetch/src/query.rs
Comment (anthony-near) [reply near-fetch/src/query.rs:602]: The `wait_until` should be an optional param to pass through with Executed as the default

Comment (anthony-near) [reply near-fetch/src/query.rs:602]: This will require that you match on multiple return types below, since for `INCLUDED` and maybe `INCLUDED_FINAL` the txn hash is returned.

Comment (anthony-near) [reply near-fetch/src/query.rs:602]: I think you also might need to update the rpc call in other parts of the code since this only queries the transaction status. We'll also need to modify it for actually sending the txn

Comment (r-near) [reply near-fetch/src/query.rs:602]: >We'll also need to modify it for actually sending the txn

You're right - the compiler only complained about this function. Turns out, the `send_tx` methods are using the old `broadcast_tx` methods on the RPC.

Comment (anthony-near) [reply near-fetch/src/query.rs:602]: The updated code should be based on https://github.com/near/near-jsonrpc-client-rs/blob/master/examples/send_tx.rs#L62

File: near-fetch/src/lib.rs
Comment (r-near) [inline near-fetch/src/lib.rs:129]: Is this right? `map_err`? Ran into a ton of issues getting this to compile correctly.

File: near-fetch/src/lib.rs
Comment (r-near) [inline near-fetch/src/lib.rs:292]: Is there a cleaner way to do this? Ran into some weirdness with references and borrowing. Also I changed the function signature which might be a no-no

File: near-fetch/src/ops.rs
Comment (r-near) [inline near-fetch/src/ops.rs:526]: Should this be `Executed` for wait_until here?

File: near-fetch/src/lib.rs
Comment (r-near) [inline near-fetch/src/lib.rs:155]: This isn't actually true. It should be Execution Status of None. But when I do that, the response doesn't come back with a TX hash. Not sure how to rectify this.

**Comment by DavidM-D** [comment]
Anything I can do to help move this along @r-near ?

**Comment by r-near** [comment]
> Anything I can do to help move this along @r-near ?

Yes definitely! Could you review the PR and approve/merge it? I don't have write access.

**Review by DavidM-D — APPROVED** [review]
Looks mostly fine to me, we can merge this when @ChaoticTempest is back. Could be worth moving this into the NEAR namespace if he's alright with it.

File: near-fetch/src/lib.rs
Comment (DavidM-D) [inline near-fetch/src/lib.rs:290]: What is this unwrapping? Is it definitely going to succeed?

File: near-fetch/src/ops.rs
Comment (ChaoticTempest) [reply near-fetch/src/ops.rs:538]: It's quite unclear how the `wait_until` parameter interacts with `RpcTransactionStatusRequest`. Let me see if I can get someone from protocol to comment on this

File: near-fetch/src/lib.rs
Comment (ChaoticTempest) [reply near-fetch/src/lib.rs:290]: yeah, this is gonna fail on the case that the tx hasn't been finalized yet. Look at the next comment for how this should be fixed

File: near-fetch/src/lib.rs
Comment (ChaoticTempest) [reply near-fetch/src/lib.rs:305]: One thing is that the option needs to be explicitly handled, and we don't need to clone here to go into an `into_outcome`. We can directly reference the members of the enum themselves and get the outcome as a reference. And with this, we do not need to change the function signature at all nor need to clone the errors at all:

```rs
async fn fetch_tx_errs(result: &RpcTransactionResponse) -> Vec<&TxExecutionError> {
    let mut failures = Vec::new();
    let Some(outcome) = result.final_execution_outcome.as_ref() else {
        return failures;
    };
    let outcome = match outcome {
        FinalExecutionOutcomeViewEnum::FinalExecutionOutcome(outcome) => outcome,
        FinalExecutionOutcomeViewEnum::FinalExecutionOutcomeWithReceipt(outcome) => {
            &outcome.final_outcome
        }
    };

    if let FinalExecutionStatus::Failure(tx_err) = &outcome.status {
        failures.push(tx_err);
    }
    if let ExecutionStatusView::Failure(tx_err) = &outcome.transaction_outcome.outcome.status {
        failures.push(tx_err);
    }
    for receipt in &outcome.receipts_outcome {
        if let ExecutionStatusView::Failure(tx_err) = &receipt.outcome.status {
            failures.push(tx_err);
        }
    }
    failures
}
```

**Review by ChaoticTempest — CHANGES_REQUESTED** [review]
This is moving in the right direction and looking like it's almost ready to be merged besides a couple key things. Let's fix those first so we don't run into any hiccups

**Comment by ChaoticTempest** [comment]
> Could be worth moving this into the NEAR namespace if he's alright with it.

yup, I'm fine with moving this into NEAR namespace after we get this PR merged

File: near-fetch/src/query.rs
Comment (r-near) [reply near-fetch/src/query.rs:594]: I'll defer to you for how best to implement this. For what it's worth, part of the motivation for updating the dependencies here was to use the `wait_until` param in downstream applications (specifically the [meta-transaction relayer](https://github.com/near/pagoda-relayer-rs))

**Pull request merged**
